exp/controllers/azuremanagedcontrolplane_controller.go

```diff
@@ -225,13 +179,11 @@ func (r *AzureManagedControlPlaneReconciler) Reconcile(ctx context.Context, req
 
 	// Create the scope.
 	mcpScope, err := scope.NewManagedControlPlaneScope(ctx, scope.ManagedControlPlaneScopeParams{
-		Client:           r.Client,
-		Logger:           log,
-		Cluster:          cluster,
-		ControlPlane:     azureControlPlane,
-		MachinePool:      ownerPool,
-		InfraMachinePool: defaultPool,
```

Comment:
I think we can remove InfraMachinePool from ManagedControlPlaneScope now?

Reply:
you've done it here, but I think it can be removed from the struct definitions as well

Reply:
I did it remove it initially, but then had to add it back as the ammp controller also uses ManagedControlPlaneScope. The InfraMachinePool is required here.